core/src/software/aws/toolkits/core/ToolkitClientManager.kt

```diff
@@ -97,37 +99,50 @@ abstract class ToolkitClientManager(private val sdkHttpClient: SdkHttpClient) {
      * Creates a new client for the requested [AwsClientKey]
      */
     @Suppress("UNCHECKED_CAST")
-    protected open fun <T : SdkClient> createNewClient(key: AwsClientKey, region: AwsRegion = key.region, credProvider: ToolkitCredentialsProvider = getCredentialsProvider()): T {
-        val builderMethod = key.serviceClass.java.methods.find {
-            it.name == "builder" && Modifier.isStatic(it.modifiers) && Modifier.isPublic(it.modifiers)
-        } ?: throw IllegalArgumentException("Expected service interface to have a public static `builder()` method.")
-        val builder = builderMethod.invoke(null) as AwsDefaultClientBuilder<*, *>
-
-        return builder
-            .httpClient(sdkHttpClient)
-            .credentialsProvider(credProvider)
-            .region(Region.of(region.id))
-            .overrideConfiguration {
-                it.putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, userAgent)
-                if (builder is S3ClientBuilder) {
-                    // TODO: Remove after SDK code-gens these instead of uses class loader
-                    it.addExecutionInterceptor(EndpointAddressInterceptor())
-                    it.addExecutionInterceptor(CreateBucketInterceptor())
-                    it.addExecutionInterceptor(PutObjectInterceptor())
-                    it.addExecutionInterceptor(EnableChunkedEncodingInterceptor())
-                    it.addExecutionInterceptor(DisableDoubleUrlEncodingInterceptor())
-                    it.addExecutionInterceptor(DecodeUrlEncodedResponseInterceptor())
-                }
-            }
-            .also { _ ->
-                if (builder is S3ClientBuilder) {
-                    builder.serviceConfiguration { it.pathStyleAccessEnabled(true) }
-                }
-            }
-            .build() as T
-    }
+    protected open fun <T : SdkClient> createNewClient(
+        key: AwsClientKey,
+        region: AwsRegion = key.region,
+        credProvider: ToolkitCredentialsProvider = getCredentialsProvider()
+    ): T = createNewClient(key.serviceClass as KClass<T>, sdkHttpClient, Region.of(region.id), credProvider)
 
     companion object {
         private val GLOBAL_SERVICES = setOf("IamClient")
+
+        fun <T : SdkClient> createNewClient(
```

Comment:
Should the telemetry client also use the new util?

https://github.com/aws/aws-toolkit-jetbrains/blob/8acb5d3f1d59df1caa2ec1223107f66d71c135ff/jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt#L71

Reply:
I think so but looks like we will need to add endpoint to the method